---
title: "biomass_compare"
author: "Nathan R. Geraldi"
date: "April 6, 2019"
output: github_document
---

set table options
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## libraries
```{r libraries, message=FALSE, warning=FALSE}
library(RColorBrewer)
library(tidyverse) 
library(broom)
#library(edgeR)
library(corrplot)
library(lme4)
library(lmerTest)
library(scales)
library(ggplot2)

library(spdep)
library(spatialreg)

#  install.packages("devtools")
#  devtools::install_github("grunwaldlab/metacoder")

```

## functions
```{r functions, message=FALSE, warning=FALSE}
# function to remove rows with n number of NA's
delete.na <- function(DF, n=0) {
  DF[rowSums(is.na(DF)) <= n,]
}

# Specify alpha as a percentage:
colorRampAlpha <- function(..., n, alpha) {
   colors <- colorRampPalette(...)(n)
   paste(colors, sprintf("%x", ceiling(255*alpha)), sep="")
}

# for good tick mark labels for ggplot
base_breaks <- function(n = 10){
    function(x) {
        axisTicks(log10(range(x, na.rm = TRUE)), log = TRUE, n = n)
    }
}
## to remove attribures from data

one_entry <- function(x) {
  for (i in length(x)) attr(x[[i]], "names") <- NULL
  return(x)
}
```


## define universal variables
```{r define_universal}
stud_pat<-"Dammam"  # matches study specific title from pipe (begining of files).
dir<-"/Users/geraldn/Dropbox/"
out_file<-"Documents/KAUST/eDNA/R/pipe_summary"
# export  to project folder
export_file<-"Documents/KAUST/eDNA/R/csv/"

```

## import taxa t
```{r taxa_table}
## taxa conversions to make data comparable
taxa_DNA_net<-openxlsx::read.xlsx(xlsxFile="/Users/geraldn/Dropbox/Global_databases/DMAP/tow_biomass/taxa_table.xlsx", sheet=1)
taxa_net_DNA<-openxlsx::read.xlsx(xlsxFile="/Users/geraldn/Dropbox/Global_databases/DMAP/tow_biomass/taxa_table.xlsx", sheet=2)
```


## import geo
```{r taxa_table}
# get ocean data  - used in "tidy 18s"
geo<-read.table("/Users/geraldn/Dropbox/Documents/KAUST/eDNA/DMAP/R/CSV/Global_layers_genome.csv", sep=",",header=T)
#  names(geo)
geo2<- geo %>% 
  dplyr::select(Latitude:ocean) %>% 
  distinct(Latitude,Latitude, .keep_all = T)
  
  
```


## metagenome
```{r metagenome}

# genome data from DMAP        
dat_genome<-data.table::fread(file = '/Users/geraldn/Dropbox/Documents/KAUST/eDNA/DMAP/R/CSV/DMAP_biomass_apr19.csv', sep = ',', header = TRUE)
##   add column differentiat malaspina profile and deep
dat_genome1<- dat_genome %>%   # names(dat_genome)      
  mutate(pid=forcats::fct_recode(pid, "PID_50" ="pid50","PID_70" ="pid70","PID_90" ="pid90" )) %>% 
  mutate(Depth = ifelse(Cruise == "TARA", Depth.env, Depth)) %>% 
  group_by_at(vars(unique_ID:pid, Latitude,Longitude,Depth,phylum)) %>% # was sp_num in stats_source
  summarize(reads=sum(reads)) %>% 
  ungroup() %>% 
  group_by_at(vars(unique_ID:pid)) %>% 
  mutate(reads_sample=sum(reads)) %>% 
  ungroup() %>% 
  mutate(reads_percent=reads/reads_sample) %>% 
  mutate(reads_percent=reads_percent*100) %>%
  dplyr::rename(Phylum=phylum) 
  # unique(dat_genome1$pid)
  ## get oceans, but need to remove some samples without lat/lon
 dat_genome_na<- dat_genome1 %>% # remove then bind back in after join-erro with na's
    filter(is.na(Latitude))

 dat_genome2<- dat_genome1 %>% # names(dat_genome2)
    filter(complete.cases(Latitude)) %>% 
    fuzzyjoin::geo_left_join(geo2, max_dist = 0.1)  %>% #  to get oceans, dist in miles
    dplyr::select(-Latitude.y,  -Longitude.y) %>% 
    dplyr::rename(Latitude=Latitude.x , Longitude=Longitude.x)  %>% 
    mutate(ocean=as.character(ocean)) %>%  #  unique(dat_genome2$ocean)
    mutate(ocean=replace(ocean, ocean=="Antarctic","Southern Ocean")) %>% 
    mutate(Latitude_abs=abs(Latitude)) %>% 
    mutate(Latitude_cat=cut(Latitude_abs, breaks=c(-Inf,23.5,35, Inf), 
                        labels=c("Tropical","Subtropical","Temperate"))) %>% 
   #bind_rows(dat_genome_na) %>% 
   # remove NaN- only for samples with no hits
   filter(!reads_percent=="NaN") 

 #########
  #       get filter catagories
filcat<-data.frame(unique(dat_genome2$filter_size))
colnames(filcat)[1] <- "filter_size"
filcat$filter_size_cat<-c("0.2-3","1-20","1-20","0.2-3","0.2-3","1-20","0.1-0.22","0.2-3","0.2-3","0.2-3")
dat_genome2<-left_join(dat_genome2, filcat, by="filter_size")   #  unique(dat_genome2$filter_size_cat)
##  get depth catagories
dat_genome2$Depth_cat <- cut(dat_genome2$Depth, breaks=c(-Inf,11, 200, 1000, 2000, 3910, Inf), 
                      labels=c("<10m","10-200m","200-1000m","1000-2000m","2000-3910m","3995-4020m"))
##
## limit to "good data"
dat_genome3<- dat_genome2 %>%   # names(dat_genome2)
  filter(gene_domain=="ec4")  %>%   # unique(dat_genome2$gene_domain)
   # limit to target DMAP filter
  filter(pid=="PID_70") %>%  # 
    ###
    filter(filter_size_cat=="0.2-3" ) 
  
dat_genome3<-data.frame(dat_genome3)

# get reads per phylum   names(dat_genome3)
dat_genome4 <- dat_genome3 %>% 
  select(-Phylum,-reads,-reads_percent,-number_of_) %>% 
  distinct()
  #filter(!duplicated(unique_ID,ID.x,Sample,Station,Cruise,area,filter_size,Latitude,Longitude,Depth))


```


## tidy
```{r tidy}
# split up by areas and standardize

sdat <-dat_genome4 %>% 
  filter(Cruise=="Malaspina" & area=="deep") 
sdat$sDepth<-scale(sdat$Depth)
sdat$sLand_Dist<-scale(sdat$land_dist)
sdat$sLatitude<-scale(sdat$Latitude)
sdat$sLatitude_abs<-scale(sdat$Latitude_abs)
sdat$sPresent.Surface.Temperature.Mean<-scale(sdat$Present.Surface.Temperature.Mean)
sdat$sPresent.Surface.Temperature.Range<-scale(sdat$Present.Surface.Temperature.Range)
sdat$sPresent.Surface.Primary.productivity.Mean<-scale(sdat$Present.Surface.Primary.productivity.Mean)
sdat$sPresent.Surface.Primary.productivity.Range<-scale(sdat$Present.Surface.Primary.productivity.Range)
sdat$sPop_in_100km.Mean<-scale(sdat$Pop_in_100km)
sdat$sPop_in_1000km<-scale(sdat$Pop_in_1000km)
sdat$sOHI_2013<-scale(sdat$OHI_2013)
deep_att<-sdat
sdat<-lapply(sdat, FUN=one_entry)
deep<-data.frame(sdat)


sdat <-dat_genome4 %>% 
  filter(Cruise=="Malaspina" & area=="prof")
sdat$sDepth<-scale(sdat$Depth)
sdat$sLand_Dist<-scale(sdat$land_dist)
sdat$sLatitude<-scale(sdat$Latitude)
sdat$sLatitude_abs<-scale(sdat$Latitude_abs)
sdat$sPresent.Surface.Temperature.Mean<-scale(sdat$Present.Surface.Temperature.Mean)
sdat$sPresent.Surface.Temperature.Range<-scale(sdat$Present.Surface.Temperature.Range)
sdat$sPresent.Surface.Primary.productivity.Mean<-scale(sdat$Present.Surface.Primary.productivity.Mean)
sdat$sPresent.Surface.Primary.productivity.Range<-scale(sdat$Present.Surface.Primary.productivity.Range)
sdat$sPop_in_100km.Mean<-scale(sdat$Pop_in_100km)
sdat$sPop_in_1000km<-scale(sdat$Pop_in_1000km)
sdat$sOHI_2013<-scale(sdat$OHI_2013)
prof_att<-sdat
sdat<-lapply(sdat, FUN=one_entry)
prof<-data.frame(sdat)

sdat <-dat_genome4 %>% 
  filter(Cruise=="TARA" )
sdat$sDepth<-scale(sdat$Depth)
sdat$sLand_Dist<-scale(sdat$land_dist)
sdat$sLatitude<-scale(sdat$Latitude)
sdat$sLatitude_abs<-scale(sdat$Latitude_abs)
sdat$sPresent.Surface.Temperature.Mean<-scale(sdat$Present.Surface.Temperature.Mean)
sdat$sPresent.Surface.Temperature.Range<-scale(sdat$Present.Surface.Temperature.Range)
sdat$sPresent.Surface.Primary.productivity.Mean<-scale(sdat$Present.Surface.Primary.productivity.Mean)
sdat$sPresent.Surface.Primary.productivity.Range<-scale(sdat$Present.Surface.Primary.productivity.Range)
sdat$sPop_in_100km.Mean<-scale(sdat$Pop_in_100km)
sdat$sPop_in_1000km<-scale(sdat$Pop_in_1000km)
sdat$sOHI_2013<-scale(sdat$OHI_2013)
TARA_att<-sdat
sdat<-lapply(sdat, FUN=one_entry)
TARA<-data.frame(sdat)

## check distribution
x<-sdat$reads_sample    # sqrt best    profile only -raw
hist(x)
hist(sqrt(x))
hist(log(x))
```


## stats tara
```{r stats tara}
#   names(DNA_all)
sdat <- TARA  %>% 
    mutate(resp=log(reads_sample+1))  # hist(sdat$resp)

############################################################################################################
########################################################################################################
####   spatial stats  names(sdat)
#Create a k=4 nearest neighbor set
sdat<-sdat[complete.cases(sdat$Latitude),]
ssdat<-sdat
sp::coordinates(ssdat) <- ~Longitude+Latitude
#
us.nb4<-knearneigh(coordinates(ssdat), k=8)
us.nb4<-knn2nb(us.nb4)
us.wt4<-nb2listw(us.nb4, style="W")
#
########################################################################################################
########################################################################################################
########################################################################################################
### names(sdat)     names(sdat2) head

lmer.mod<-lmer(resp~sDepth+sLand_Dist+sPresent.Surface.Temperature.Mean+
    sPresent.Surface.Temperature.Range+sPresent.Surface.Primary.productivity.Mean+
    sPresent.Surface.Primary.productivity.Range+sOHI_2013+(1|lohghurst_biome/Station),data=sdat)##
#
#   mes<- c(AIC(lmer.mod),AIC(lmer.mod2))
mod<-lm(resp~sDepth+sLand_Dist+sPresent.Surface.Temperature.Mean+
          sPresent.Surface.Temperature.Range+sPresent.Surface.Primary.productivity.Mean+
          sPresent.Surface.Primary.productivity.Range+sOHI_2013,ssdat)
#SAR - Lag model
fit.lag<-lagsarlm(mod, ssdat, listw=us.wt4, type="lag", method="MC")
#  summary(fit.lag, Nagelkerke=T)
#SAR - Error model
fit.err<-errorsarlm(mod, ssdat, listw=us.wt4, etype="error", method="MC")
#    summary(fit.err, Nagelkerke=T)
#Spatial Durbin Model
fit.durb<-lagsarlm(mod, ssdat, listw=us.wt4, type="mixed", method="MC")
#   summary(fit.durb, Nagelkerke=T)
#Spatial Durbin Error Model
fit.errdurb<-errorsarlm(mod, ssdat, listw=us.wt4, etype="emixed", method="MC")
#   summary(fit.errdurb, Nagelkerke=T)
#SAC Model  
fit.sac<-sacsarlm(mod, ssdat, listw=us.wt4, type="sac", method="MC")
#    summary(fit.sac, Nagelkerke=T)
#SMA model  
fit.sma<-spautolm(mod, ssdat, listw=us.wt4, family="SMA")
#   summary(fit.sma, Nagelkerke=T)
AICs<-c(AIC(lmer.mod),AIC(mod),AIC(fit.lag), AIC(fit.err), AIC(fit.durb), AIC(fit.errdurb), AIC(fit.sac), AIC(fit.sma))
AICs
#  fit.lag  #  best
###################  test interactions
#           #   summary(fit.lag, Nagelkerke=T),    summary(mod)
### check if interactions improve model

mod2<-lm(resp~(sDepth+sLand_Dist+sPresent.Surface.Temperature.Mean+
                 sPresent.Surface.Temperature.Range+sPresent.Surface.Primary.productivity.Mean+
                 sPresent.Surface.Primary.productivity.Range+sOHI_2013)^2,data=ssdat)##

mod2<-lm(resp~sDepth+sLand_Dist+sPresent.Surface.Temperature.Mean+
                         sPresent.Surface.Temperature.Range+sPresent.Surface.Primary.productivity.Mean+
                         sPresent.Surface.Primary.productivity.Range+sOHI_2013
           
           ,data=ssdat)

fit.lag<-lagsarlm(mod2, ssdat, listw=us.wt4, type="lag", method="MC")
summary(mod2)
AIC(fit.sma)   ## 790 aic 

##check assumptions
mod<-fit.lag   # mod<-mod2
hist(resid(mod))
plot(fitted(mod),resid(mod))  #
abline(0,0)
qqnorm(resid(mod))
qqline(resid(mod)) 
```


## stats mal prof
```{r stats prof}
#   names(DNA_all)
sdat <- prof %>% 
    mutate(resp=log(reads_sample+1))  # hist(sdat$resp)    hist(sdat$reads_sample)

############################################################################################################
########################################################################################################
####   spatial stats  names(sdat)
#Create a k=4 nearest neighbor set
sdat<-sdat[complete.cases(sdat$Latitude),]
ssdat<-sdat
sp::coordinates(ssdat) <- ~Longitude+Latitude
#
us.nb4<-knearneigh(coordinates(ssdat), k=8)
us.nb4<-knn2nb(us.nb4)
us.wt4<-nb2listw(us.nb4, style="W")
#
########################################################################################################
########################################################################################################
########################################################################################################
###  names(sdat)     names(sdat2) head

lmer.mod<-lmer(resp~sDepth+sLand_Dist+sPresent.Surface.Temperature.Mean+
    sPresent.Surface.Temperature.Range+sPresent.Surface.Primary.productivity.Mean+
    sPresent.Surface.Primary.productivity.Range+sOHI_2013+(1|lohghurst_biome/Station),data=sdat)##
#  summary(lmer.mod)
#   mes<- c(AIC(lmer.mod),AIC(lmer.mod2))
mod<-lm(resp~sDepth+sLand_Dist+sPresent.Surface.Temperature.Mean+
          sPresent.Surface.Temperature.Range+sPresent.Surface.Primary.productivity.Mean+
          sPresent.Surface.Primary.productivity.Range+sOHI_2013,ssdat)
#SAR - Lag model
fit.lag<-lagsarlm(mod, ssdat, listw=us.wt4, type="lag", method="MC")
#  summary(fit.lag, Nagelkerke=T)
#SAR - Error model
fit.err<-errorsarlm(mod, ssdat, listw=us.wt4, etype="error", method="MC")
#    summary(fit.err, Nagelkerke=T)
#Spatial Durbin Model
fit.durb<-lagsarlm(mod, ssdat, listw=us.wt4, type="mixed", method="MC")
#   summary(fit.durb, Nagelkerke=T)
#Spatial Durbin Error Model
fit.errdurb<-errorsarlm(mod, ssdat, listw=us.wt4, etype="emixed", method="MC")
#   summary(fit.errdurb, Nagelkerke=T)
#SAC Model  
fit.sac<-sacsarlm(mod, ssdat, listw=us.wt4, type="sac", method="MC")
#    summary(fit.sac, Nagelkerke=T)
#SMA model  
fit.sma<-spautolm(mod, ssdat, listw=us.wt4, family="SMA")
#   summary(fit.sma, Nagelkerke=T)
AICs<-c(AIC(lmer.mod),AIC(mod),AIC(fit.lag), AIC(fit.err), AIC(fit.durb), AIC(fit.errdurb), AIC(fit.sac), AIC(fit.sma))
AICs
#  fit.sma  #  best but very bad fit, next was lmer.mod
###################  test interactions
#           #   summary(fit.sma, Nagelkerke=T)   summary(lmer.mod)
### check if interactions improve model

mod2<-lmer(resp~(sDepth+sLand_Dist+sPresent.Surface.Temperature.Mean+
    sPresent.Surface.Temperature.Range+sPresent.Surface.Primary.productivity.Mean+
    sPresent.Surface.Primary.productivity.Range+sOHI_2013)^2+(1|lohghurst_biome/Station),data=sdat)##

mod2<-lmer(resp~sDepth+sLand_Dist+sPresent.Surface.Temperature.Mean+
    sPresent.Surface.Temperature.Range+sPresent.Surface.Primary.productivity.Mean+
    sPresent.Surface.Primary.productivity.Range+sOHI_2013
    +(1|lohghurst_biome/Station),data=sdat)##

fit.sma<-spautolm(mod2, ssdat, listw=us.wt4, family="SMA")
summary(mod2)
AIC(fit.sma)   ## 790 aic 
##check assumptions
mod<-mod2
hist(resid(mod))
plot(fitted(mod),resid(mod))  ## this is very bad !!!!!!   use 2nd best AIC- mod
abline(0,0)
qqnorm(resid(mod))
qqline(resid(mod))
```


## stats mal deep
```{r stats deep}
#   names(DNA_all)
sdat <- deep %>% 
  mutate(resp=log(reads_sample+1))  # hist(sdat$resp)    hist(sdat$reads_sample)

############################################################################################################
########################################################################################################
####   spatial stats  names(sdat)
#Create a k=4 nearest neighbor set
sdat<-sdat[complete.cases(sdat$Latitude),]
ssdat<-sdat
sp::coordinates(ssdat) <- ~Longitude+Latitude
#
us.nb4<-knearneigh(coordinates(ssdat), k=8)
us.nb4<-knn2nb(us.nb4)
us.wt4<-nb2listw(us.nb4, style="W")
#
########################################################################################################
########################################################################################################
########################################################################################################
###  names(sdat)     names(sdat2) head

lmer.mod<-lmer(resp~sDepth+sLand_Dist+sPresent.Surface.Temperature.Mean+
    sPresent.Surface.Temperature.Range+sPresent.Surface.Primary.productivity.Mean+
    sPresent.Surface.Primary.productivity.Range+sOHI_2013+(1|lohghurst_biome/Station),data=sdat)##
#
#   mes<- c(AIC(lmer.mod),AIC(lmer.mod2))
mod<-lm(resp~sDepth+sLand_Dist+sPresent.Surface.Temperature.Mean+
          sPresent.Surface.Temperature.Range+sPresent.Surface.Primary.productivity.Mean+
          sPresent.Surface.Primary.productivity.Range+sOHI_2013,ssdat)
#SAR - Lag model
fit.lag<-lagsarlm(mod, ssdat, listw=us.wt4, type="lag", method="MC")
#  summary(fit.lag, Nagelkerke=T)
#SAR - Error model
fit.err<-errorsarlm(mod, ssdat, listw=us.wt4, etype="error", method="MC")
#    summary(fit.err, Nagelkerke=T)
#Spatial Durbin Model
fit.durb<-lagsarlm(mod, ssdat, listw=us.wt4, type="mixed", method="MC")
#   summary(fit.durb, Nagelkerke=T)
#Spatial Durbin Error Model
fit.errdurb<-errorsarlm(mod, ssdat, listw=us.wt4, etype="emixed", method="MC")
#   summary(fit.errdurb, Nagelkerke=T)
#SAC Model  
fit.sac<-sacsarlm(mod, ssdat, listw=us.wt4, type="sac", method="MC")
#    summary(fit.sac, Nagelkerke=T)
#SMA model  
fit.sma<-spautolm(mod, ssdat, listw=us.wt4, family="SMA")
#   summary(fit.sma, Nagelkerke=T)
AICs<-c(AIC(lmer.mod),AIC(mod),AIC(fit.lag), AIC(fit.err), AIC(fit.durb), AIC(fit.errdurb), AIC(fit.sac), AIC(fit.sma))
AICs
#  fit.sma  #  best
###################  test interactions
#           #   summary(fit.sma, Nagelkerke=T)
### check if interactions improve model

mod2<-lm(resp~(sDepth+sLand_Dist+sPresent.Surface.Temperature.Mean+
                 sPresent.Surface.Temperature.Range+sPresent.Surface.Primary.productivity.Mean+
                 sPresent.Surface.Primary.productivity.Range+sOHI_2013)^2,data=ssdat)##

mod2<-lm(resp~sDepth+sLand_Dist+sPresent.Surface.Temperature.Mean+
                         sPresent.Surface.Temperature.Range+sPresent.Surface.Primary.productivity.Mean+
                         sPresent.Surface.Primary.productivity.Range+sOHI_2013
                                
      #  Not enough replicates to run interations !!!!!!!
           ,data=ssdat)

fit.sma<-spautolm(mod2, ssdat, listw=us.wt4, family="SMA")
summary(fit.sma, Nagelkerke=T)
AIC(fit.sma)   ## 790 aic 
plot(fit.sma)

########################################################################################################
##check assumptions
mod<-fit.sma
hist(resid(mod))
plot(fitted(mod),resid(mod))
abline(0,0)
qqnorm(resid(mod))
qqline(resid(mod))
plot(mod)###residuals should be random (x=fitted value, y=standardixed res)
plot(mod@frame$sqrt(reads)~fitted(mod))
plot(md$reads,fitted(mod))###resp linear fudncion of teh fitted (x=fitted value, y=response)
qqnorm(mod~resid(mod)|md$reads)##errors are close to normally dist in blocks-linear in plots(x-residuals, y-quantiles of standard norm)
```

## plot sig TARA
```{r plot TARA}
names(DNA)
cc1<-c('#543005','#8c510a','#bf812d','#dfc27d','#f6e8c3','#c7eae5','#80cdc1','#35978f','#01665e','#003c30')
cc4<-c('#543005','#dfc27d','#80cdc1','black')
c1<-c('#543005')
lab<-c("0.5-5" , "5-20", "20-200","180-2000" )
n<-29
cc_n<- colorRampPalette(col=c("#543005", "#fee08b", "#003c30"))(n)
shap<-rep(1:6, length.out=n)
## for scale_descrete
fac_col <-setNames( cc4, lab )

####    TARA  ##
gdat <- TARA  %>% 
    mutate(resp=log(reads_sample+1)) %>%  # hist(sdat$resp) names(gdat)
 mutate(lohghurst_biome=factor(lohghurst_biome,levels=levels(dat_genome4$lohghurst_biome))) 
gdat_att<-TARA_att

## back trans y axis and get new tics and clean numbers
aa=data.frame(x=c(1:4))
aa<- aa %>% 
  mutate(real_num=x) %>% 
  mutate(lab=(exp(x)+1)) %>% 
  mutate(nlab=c(1,10,50,100)) %>% 
  mutate(ntic=log(nlab+1))
# length(unique(as.character(gdat$lohghurst_biome)))
# from model
int=1.521
slop=-0.459
## unscale x axis
hp<-as.vector(DMwR::unscale(c(0:3), gdat_att$sDepth)) # numbers from original plot
hp2<-c(10,300,600,900)
hp2_lab<-hp2 # c("5","500","10,000,000")
#hp3<-log(hp2+1)
hp4<- (hp2 - mean(gdat$Depth, na.rm=T))/ sd(gdat$Depth, na.rm=T) 

tp_depth <- ggplot(gdat, aes(x=sDepth, y=resp, color=lohghurst_biome, shape=lohghurst_biome)) + # , fill=filter_size_cat
  geom_point() +
  #geom_smooth(method = "gam") + # gam or loess
  geom_abline(intercept = int, slope = slop) +
  scale_color_manual(values=cc_n, drop = FALSE) +
  scale_shape_manual(values=shap, drop = FALSE) +
  labs(y = "Sequence abundance (per million sequences)", x = "Depth (m)")  +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())  +
  scale_x_continuous(breaks = hp4, labels = hp2_lab) +
  scale_y_continuous(breaks = aa$ntic, labels = aa$nlab) +
    theme(legend.position = "none")  

# from model
int=1.521
slop=-0.298
## unscale x axis
hp<-as.vector(DMwR::unscale(c(-4,-2.5,0.5,1), gdat_att$sPresent.Surface.Temperature.Mean)) # numbers from original plot
hp2<-c(1,10,20,30)
hp2_lab<-hp2 
hp4<- (hp2 - mean(gdat$Present.Surface.Temperature.Mean, na.rm=T))/ sd(gdat$Present.Surface.Temperature.Mean, na.rm=T) 

tp_tempm <- ggplot(gdat, aes(x=sPresent.Surface.Temperature.Mean, y=resp, color=lohghurst_biome, shape=lohghurst_biome)) + # , fill=filter_size_cat
  geom_point() +
  #geom_smooth(method = "gam") + # gam or loess
  geom_abline(intercept = int, slope = slop) +
  scale_color_manual(values=cc_n, drop = FALSE) +
  scale_shape_manual(values=shap, drop = FALSE) +
  labs(y = "Sequence abundance (per million sequences)", x = "Mean SST (°C)")  + 
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())  +
  scale_x_continuous(breaks = hp4, labels = hp2_lab) +
  scale_y_continuous(breaks = aa$ntic, labels = aa$nlab) +
    theme(legend.position = "none")  
```
 

## plot sig ?
```{r plot prof}

# mala profile  
gdat <- prof  %>% 
    mutate(resp=log(reads_sample+1)) %>%  # hist(sdat$resp) names(gdat)
    mutate(lohghurst_biome=factor(lohghurst_biome,levels=levels(dat_genome4$lohghurst_biome))) # #  unique(gdat$lohghurst_biome)
gdat_att<-prof_att

## back trans y axis and get new tics and clean numbers
aa=data.frame(x=c(3:6))
aa<- aa %>% 
  mutate(real_num=x) %>% 
  mutate(lab=(exp(x)+1)) %>% 
  mutate(nlab=c(25,50,150,400)) %>% 
  mutate(ntic=log(nlab+1))
# length(unique(as.character(gdat$lohghurst_biome)))

##  Depth  #####
# from model
int=4.67
slop=0.0522
## unscale x axis
hp<-as.vector(DMwR::unscale(c(-0.9:2), gdat_att$sDepth)) # numbers from original plot
hp2<-c(5,1000,2000,3000, 4000)
hp2_lab<-hp2 # c("5","500","10,000,000")
#hp3<-log(hp2+1)
hp4<- (hp2 - mean(gdat$Depth, na.rm=T))/ sd(gdat$Depth, na.rm=T) 

prof_depth <- ggplot(gdat, aes(x=sDepth, y=resp, color=lohghurst_biome, shape=lohghurst_biome)) + #
  geom_point() +
  #geom_smooth(method = "gam") + # gam or loess
  geom_abline(intercept = int, slope = slop) +
  scale_color_manual(values=cc_n, drop = FALSE) +
  scale_shape_manual(values=shap, drop = FALSE) +
  labs(y = "Sequence abundance (per million sequences)", x = "Depth (m)")  +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())  +
  scale_x_continuous(breaks = hp4, labels = hp2_lab) +
  scale_y_continuous(breaks = aa$ntic, labels = aa$nlab) +
    theme(legend.position = "none")  

###   mean temp   #####
# from model
int=4.67
slop=-0.00211
## unscale x axis
hp<-as.vector(DMwR::unscale(c(-4,-2.5,0.5,1), gdat_att$sPresent.Surface.Temperature.Mean)) # numbers from original plot
hp2<-c(1,10,20,30)
hp2_lab<-hp2 
hp4<- (hp2 - mean(gdat$Present.Surface.Temperature.Mean, na.rm=T))/ sd(gdat$Present.Surface.Temperature.Mean, na.rm=T) 

prof_tempm <- ggplot(gdat, aes(x=sPresent.Surface.Temperature.Mean, y=resp, color=lohghurst_biome, shape=lohghurst_biome)) + # , fill=filter_size_cat
  geom_point() +
  #geom_smooth(method = "gam") + # gam or loess
  geom_abline(intercept = int, slope = slop) +
  scale_color_manual(values=cc_n) +
  scale_shape_manual(values=shap) +
  labs(y = "Sequence abundance (per million sequences)", x = "Mean SST (°C)")  + 
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())  +
  scale_x_continuous(breaks = hp4, labels = hp2_lab) +
  scale_y_continuous(breaks = aa$ntic, labels = aa$nlab) +
    theme(legend.position = "none")  
#print(dpp)
## combine
p<- ggpubr::ggarrange(tp, mpp, dpp, #+ ggpubr::rremove("x.text"), 
                  labels = c("A", "B", "C"),
                  font.label = list(size = 10, face = "plain"),
                  ncol = 1, nrow = 3,
                  hjust=-5, vjust=1, align= "hv")  # smaler - right, smaller up
                  #common.legend = TRUE, legend = "bottom")
  print(p)
## save  
ggsave(filename="/Users/geraldn/Dropbox/Documents/KAUST/eDNA/DMAP/R/plots/comparison_2019/rich_abund_3cruises.pdf"
        , plot = p, width = 20, height = 30, units = "cm" )

```











